mapcrypto.c

```diff
@@ -673,6 +684,199 @@ void msDecryptStringWithDPAPI( const char * in, char * out )
    *out = '\0';
 }
 
+/**********************************************************************
+*                        msEncryptStringWithDPAPING()
+*
+* Encrypts and base64-encodes the contents of string in[] using certificate and returns the
+* result in **out. Caller must free up the memory allocated by out.
+**********************************************************************/
+int msEncryptStringWithDPAPING( const char * thumbprint, const char *in, char ** out )
+{
+   DATA_BLOB DataEncrypted, DataUnencrypted;
+   BOOL result;
+   NCRYPT_DESCRIPTOR_HANDLE descriptorHandle = NULL;
+   char* descriptor;
+   const char * prefix = "CERTIFICATE=HashId:";
+
+   if( thumbprint == NULL )
+   {
+      msSetError( MS_MISCERR,
+         "Require certifcate thumbprint to encrypt the data. The certificate must be stored in the Current user stored",
+         "msEncryptStringWithDPAPING()" );
+      return MS_FAILURE;
+   }
+
+   //Set the protection descriptor string with the thumbprint;
+   descriptor = (char*)msSmallMalloc( sizeof( char ) * (strlen( thumbprint ) + strlen( prefix )) );
```

Comment:
I would like to add an extra space for the trailing 0 here.

Reply:
Where is descriptor freed?

Reply:
it gets freed by mbsrtowcs